Only do crowfly if we leave the requested area

kinnou02 opened the pull request:
- If the stop_point is part of the stop_area, then : crow_fly section, no path and no geoJson
- If the stop_point is not part of the stop_area, then : street_network section, with path and geoJson to the centroid of the stop_area

Same principle for admin.

refer to: http://jira.canaltp.fr/browse/NAVITIAII-1093

kinnou02 committed:
Kraken: only do crowfly if the used stop point is in the requested area

So if the journey start from a stop_point from another stop_area than
the requested, we display a streetnetwork section

kinnou02 committed:
Jormungandr: for crowfly section we don't create a geojson

antoine-de commented on source/routing/raptor_api.cpp:
```diff
@@ -83,16 +103,22 @@ pbnavitia::Response make_pathes(const std::vector<navitia::routing::Path>& paths
+         **/
 
-        if (origin.type == type::Type_e::Admin || origin.type == type::Type_e::StopArea) {
+        if (!path.items.front().stop_points.empty() && use_crow_fly(origin, path.items.front().stop_points.front())){
```
you can delete the test below, useless now

kinnou02 replied:
right, I have deleted them

antoine-de commented on source/routing/raptor_api.cpp:
```diff
@@ -235,7 +249,7 @@ pbnavitia::Response make_pathes(const std::vector<navitia::routing::Path>& paths
         }
 
-        if (destination.type == type::Type_e::Admin || destination.type == type::Type_e::StopArea) {
+        if (!path.items.back().stop_points.empty() && use_crow_fly(destination, path.items.back().stop_points.back())){
```
same here

kinnou02 committed:
Kraken: refactoring for add test on "use_crow_fly"

Add some forward declaration in raptor API, so I have to add header in a
lot of files

kinnou02 committed:
Kraken: add test on use_crow_fly
